From d5691b30ec9c624f97b4e58a72a2c89d3230c27e Mon Sep 17 00:00:00 2001
From: Alexandre Torgue <alexandre.torgue@foss.st.com>
Date: Wed, 7 Oct 2020 15:31:50 +0200
Subject: [PATCH 0043/1141] arm64: introduce STM32 family on ARMv8 architecture

Add a dedicated ARCH_STM32 for STM32 SoCs config. First STM32 ARMv8 SoC
family is the STM32MP25 which is composed by STM32MP251, STM32MP253,
STM32MP255, STM32MP257 SoCs.

Signed-off-by: Alexandre Torgue <alexandre.torgue@foss.st.com>
Change-Id: Icea9be8c8d0879fb303006c6f1028c9124f0dd5e
---
 arch/arm64/Kconfig.platforms | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

--- a/arch/arm64/Kconfig.platforms
+++ b/arch/arm64/Kconfig.platforms
@@ -279,6 +279,22 @@ config ARCH_INTEL_SOCFPGA
 	  Stratix 10 (ex. Altera), Stratix10 Software Virtual Platform,
 	  Agilex and eASIC N5X.
 
+config ARCH_STM32
+	bool "STMicroelectronics STM32 SoC Family"
+	select ARCH_HAS_RESET_CONTROLLER
+	select GPIOLIB
+	select PINCTRL
+	select RESET_CONTROLLER
+	select STM32_EXTI
+	select PINCTRL_STM32MP257
+	select COMMON_CLK_STM32MP257
+	select RESET_STM32MP1
+	select ARM_SMC_MBOX
+	select ARM_SCMI_PROTOCOL
+	select COMMON_CLK_SCMI
+	help
+	  This enables support for ARMv8 based STMicroelectronics STM32 family.
+
 config ARCH_SYNQUACER
 	bool "Socionext SynQuacer SoC Family"
 	select IRQ_FASTEOI_HIERARCHY_HANDLERS
